### PR TITLE
[radiko] Fix extractor

### PR DESCRIPTION
**Pull request by Lesmiscore**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

---

### Description of your *pull request* and other information

This PR improves debuggability of extractor, even without verbose log. (of course we should always require it!) And some other small fixes.

TODO
- [ ] ~~Direct users to clear cache in case of no formats found (#3652)~~

### CODE REVIEW

File: yt_dlp/extractor/radiko.py [hunk 1]
```diff
@@ -112,22 +112,26 @@ def _extract_formats(self, video_id, station, is_onair, ft, cursor, auth_token,
...
-                if re.match(r'^[cf]-radiko\.smartstream\.ne\.jp$', domain):
+                if re.fullmatch(r'[cf]-radiko\.smartstream\.ne\.jp', domain):
                     # Prioritize live radio vs playback based on extractor
                     sf['preference'] = 100 if is_onair else -100
                 if not is_onair and url_attrib['timefree'] == '1' and time_to_skip:
                     sf['downloader_options'] = {'ffmpeg_args': ['-ss', time_to_skip]}
             formats.extend(subformats)
 
+        if not formats:
+            self.raise_no_formats('No formats found! You should force re-authenticate using --extractor-args radiko:force_reauth or clear cache.')
```
Comment (pukkandan) [inline yt_dlp/extractor/radiko.py:133]: why can't we just automatically re-authenticate + re-extract here? That's what other similar extractors do

Comment (Lesmiscore) [reply yt_dlp/extractor/radiko.py:133]: Premium-only archives (without paying) falls here too

Comment (Lesmiscore) [reply yt_dlp/extractor/radiko.py:133]: Oh yes, I haven't implemented auth

makes sense

Comment (Lesmiscore) [reply yt_dlp/extractor/radiko.py]: Done

**Comment by pukkandan** [comment]
This is what I meant https://github.com/yt-dlp/yt-dlp/compare/master...pukkandan:radiko-fixx

Is the extractor-arg needed now?

**Comment by Lesmiscore** [comment]
That would be better
Yes, extractor-arg no longer needed

**Comment by pukkandan** [comment]
Can merge once tested

**Pull request merged**